doc/tools/Makefile.am

```diff
@@ -36,8 +36,11 @@ tools.html: $(srcdir)/tools.xml $(wildcard $(srcdir)/*.1.xml) $(wildcard $(srcdi
 			$(shell sed -n 's,.*<option>\([^<]*\)</option>.*<replaceable>.*,\1,pg' $< \
 				| sort -u | grep -- '^\-' | tr '\n' '|' | sed 's,|$$,,')," \
 		| sed "s,FILEOPTS,\
-			$(shell sed -n 's,.*<option>\([^<]*\)</option>.*<replaceable>.*filename.*,\1,pg' $< \
-				| sort -u | grep -- '^\-' | tr '\n' '|')," \
+			$(shell sed -n 's,.*<option>\([^<]*\)</option>.*<replaceable>.*\(filename\|path\).*,\1,pg' $< \
+				| sort -u | grep -- '^\-' | tr '\n' '|' | sed 's,|$$,,' | grep ^ || echo "!*"),"  \
+		| sed "s,MODULEOPTS,\
+			$(shell sed -n 's,.*<option>\([^<]*\)</option>.*<replaceable>.*mod.*,\1,pg' $< \
+				| sort -u | grep -- '^\-' | tr '\n' '|' | sed 's,|$$,,' | grep ^ || echo "!*")," \
```

Comment:
@eighthave would you mind having an other view this change with the bash completion?

Reply:
I can't really tell by reading it, shell scripts aren't really so readable.  It just needs to be tested.  I'm not using OpenSC these days, so I'm not setup to test.

Reply:
Well ... the point is that we add also completion for the filename arguments, which was partially done, but needed adjustments for tools that do not have any of these arguments (therefore the `grep ^ || echo "!*"`, which will never mach in the bash `case`.

The MODULEOPTS is just a bonus to complete on so objects (though we might need also dlls on Windows as I thing about it now).

Reply:
I've tried `pkcs15-init --options-file <Tab>`, which doesn't find a completion...

Also note that the argument for `opensc-explorer`'s `--mf` is described as *path*, which should not complete to a filename...

Reply:
Did you remove the old completion file before you rebuilt it (or `make clean` and installed it? This specific use case with `pkcs15-init --options-file` works fine for me in Fedora.

The way how it is generated always assumes some approximations. It is probably impossible to catch all the details but we can gradually adjust the filters above or manual pages if there is some ambiguity.

Reply:
That was it, thanks.

Renaming the argument in opensc-explorer should still be done, though.

Reply:
Can you think of better name for the `opensc-explorer --mf path` argument? For example `card-path`? But it would need the adjustments the the `sed` selection in above.

Or should we drop the `path` match and rename other occurrences of "path" in tools requiring filenames to `filename` for consistency?

Reply:
Yes, using `filename` everywhere should be more intuitive. I wonder if the description of the `--mf` argument includes a AID in the manual...

Reply:
Grepping through the code shows that only pkcs11-tool and westcos-tool are using the "path" and they are easy to fix and simplify the `sed` above to match only "filename" and then we do not have to care about the path in opensc-explorer.

Thanks for pointers. I pushed the related changes.